pkg/controllers/user/ingress/ingress.go

```diff
@@ -28,12 +22,14 @@ import (
 type Controller struct {
 	serviceLister v1.ServiceLister
 	services      v1.ServiceInterface
+	needNodePort  bool
 }
 
-func Register(ctx context.Context, workload *config.UserOnlyContext) {
+func Register(ctx context.Context, workload *config.UserOnlyContext, cluster *config.UserContext) {
```

Comment:
lets fetch cluster from cache instead of passing it to Register method. The context will have a clusterName, and you can use it to get the cluster